clippy_lints/src/casts/cast_slice_different_sizes.rs

```diff
@@ -56,21 +36,20 @@ pub(super) fn check(cx: &LateContext<'_>, expr: &Expr<'_>, msrv: &Option<RustcVe
                     expr.span,
                     &format!(
                         "casting between raw pointers to `[{}]` (element size {}) and `[{}]` (element size {}) does not adjust the count",
-                        from_slice_ty, from_size, to_slice_ty, to_size,
+                        start_ty.ty, from_size, end_ty.ty, to_size,
                     ),
                     |diag| {
-                        let cast_expr = match expr.peel_blocks().kind {
-                            ExprKind::Cast(cast_expr, ..) => cast_expr,
-                            _ => unreachable!("expr should be a cast as checked by expr_cast_chain_tys"),
-                        };
-                        let ptr_snippet = snippet_opt(cx, cast_expr.span).unwrap();
+                        let ptr_snippet = snippet_opt(cx, left_cast.span).unwrap();
```

Comment:
NIT:

`snipped_opt` can return `None`, instead it's better to use [`snippet`](https://doc.rust-lang.org/nightly/nightly-rustc/clippy_utils/source/fn.snippet.html), which will return a default instead of panicking. This has been fine so far, but better safe than sorry :upside_down_face:

```rs
let ptr_snippet = snippet(cx, left_cast.span, "..");
```